include/miopen/miopen.h

```diff
@@ -4701,6 +4701,106 @@ miopenReduceTensor(miopenHandle_t handle,
 /** @} */
 // CLOSEOUT TensorReduce DOXYGEN GROUP
 
+/**
+ * An miopenProblem will describe a problem for different miopen operations.
+ * For now, this is only used for convolution, but could be used for other
+ * operators in the future(such as GEMM, Pooling, etc)
+ */
+MIOPEN_DECLARE_OBJECT(miopenProblem);
+
+typedef enum
+{
+    miopenProblemDirectionForward        = 0,
+    miopenProblemDirectionBackward       = 1,
+    miopenProblemDirectionBackwardWeight = 2,
+} miopenProblemDirection_t;
+
+// Enums with no meaningful default value should probably have invalid default for fool-proofing.
+typedef enum
+{
+    miopenTensorNameInvalid  = 0,
+    miopenTensorConvolutionX = 1,
+    miopenTensorConvolutionW = 2,
+    miopenTensorConvolutionY = 3,
+} miopenTensorName_t;
```

Comment:
Name is too common. `miopenTensorAssignment_t`?

Reply:
"Assignment" would be confusing for both users and MIOpen developers.

Reply:
Maybe `miopenTensorBinding_t`? After all, here we tell which output or input of the primitive we bind the tensor to. I've always thought that a "name" is something the user can freely assign, but in this specific case this is impossible, and that confuses me...

Reply:
I see, I think something like `miopenTensorArgumentId_t`. We can use `ID` which doesn't mean arbitrary name the user gives. I think `miopenTensorBinding_t` is not very clear.